README.md

```diff
@@ -76,18 +116,207 @@ load(
+```python
+codechecker_test(
+    name = "your_codechecker_rule_name",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+Then invoke bazel:
+
+```bash
+bazel build ://your_codechecker_rule_name
+```
+
+You can find the analysis results in the `bazel-bin/` folder, on which you
+can run [`CodeChecker store`](https://github.com/Ericsson/codechecker/blob/master/docs/web/user_guide.md#store)
+or [`CodeChecker parse`](https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#parse):
+
+```bash
+CodeChecker parse bazel-bin/path/to/results
+```
+
+### Build-only CodeChecker analysis: `codechecker()`
+
+This rule is functionally equivalent to `codechecker_test` but omits the test phase where either PASS or FAIL is printed.
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker"
+)
+```
+
+### Multiplatform CodeChechecker analysis: `codechecker_suite()`
+
+This rule is functionally equivalent to `codechecker_test` but allows for running on multiple platforms via the `platforms` parameter.
+You can include and use it similarly as well:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_suite"
+)
+```
+
+### `codechecker_config()`
+
+Using the Bazel rule `codechecker_config()` you can parse a CodeChecker [configuration file](https://github.com/Ericsson/codechecker/blob/master/docs/config_file.md).
+
+First, include the rule in your BUILD file:
+
+```python
+load(
+    "@bazel_codechecker//src:codechecker.bzl",
+    "codechecker_config"
+)
+```
+
+Create a CodeChecker configuration file e.g. `config.json` (see example [here](https://github.com/Szelethus/codechecker_bazel/blob/readme_update/test/config.json)) and parse it using `codechecker_config()`. 
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    config_file = ":config.json"
+)
+```
+
+Alternatively, you can assemble a CodeChecker configuration without a config file using the rule:
+
+```python
+codechecker_config(
+    name = "your_codechecker_config",
+    analyze = [
+        "--enable=bugprone-dangling-handle",
+        "--enable=bugprone-fold-init-type",
+        "--enable=misc-non-copyable-objects",
+        "--report-hash=context-free-v2",
+    ]
+)
+```
+
+You can now configure your `codechecker`, `codechecker_suite`, `codechecker_test` and `code_checker_test` targets using the above configuration:
 
 ```python
 codechecker_test(
     name = "your_codechecker_rule_name",
+    config = "your_codechecker_config",
+    targets = [
+        "your_target",
+    ],
+)
+```
+
+### Per-file CodeChecker analysis: `code_checker_test()`
+> [!IMPORTANT]
+> The rule is still in prototype status and is subject to changes without notice.
+> You are free to experiment and report issues however!
+
+Instead of a single CodeChecker call, the bazel rule `code_checker_test()` invokes
```

Comment:
So, I dont think we should use code_checker_test() separately - it just adds confusion.
Probably we should rename code_checker.bzl to something per_file.bzl and use it in codechecker.bzl

Reply:
Created a new issue about this: https://github.com/Ericsson/codechecker_bazel/issues/31.